src/modules/currency/currencies.ts

```diff
@@ -0,0 +1,170 @@
+import BigNumber from 'bignumber.js';
+import memoize from 'memoize-one';
+
+interface ResolvedNumberFormatOptions {
+  numberingSystem: string;
+  style: string;
+  currency?: string;
+  currencyDisplay?: string;
+  minimumIntegerDigits: number;
+  minimumFractionDigits: number;
+  maximumFractionDigits: number;
+  minimumSignificantDigits?: number;
+  maximumSignificantDigits?: number;
+  useGrouping: boolean;
+}
+
+export enum CurrencyType {
+  Fiat,
+  Crypto,
+}
+
+export interface CurrencyConfig {
+  name: string;
+  code: string;
+  symbol: string;
+  locale: string;
+  type: CurrencyType;
+  getFormatterConfig?: (
+    value: BigNumber.Value
+  ) => Partial<ResolvedNumberFormatOptions>;
+  modifyParts?: (parts: Intl.NumberFormatPart[]) => Intl.NumberFormatPart[];
+}
+
+const noDecimalsConfig = { minimumFractionDigits: 0, maximumFractionDigits: 0 };
+const sixFractionalDigitsConfig = { maximumFractionDigits: 6 };
+const fourFractionalDigitsConfig = { maximumFractionDigits: 4 };
+
+const setCustomSymbol = (symbol: string) => (parts: Intl.NumberFormatPart[]) =>
+  parts
+    .filter((part) => part.type !== 'literal')
+    .map((part) =>
+      part.type === 'currency' ? { ...part, value: symbol } : part
+    );
+
+export const CURRENCIES: Record<string, CurrencyConfig> = {
+  usd: {
+    name: 'US Dollar',
+    code: 'usd',
+    symbol: '$',
+    locale: 'en-US',
+    type: CurrencyType.Fiat,
+  },
+  eth: {
+    name: 'Ether',
+    code: 'eth',
+    symbol: 'Ξ',
+    locale: 'en-US',
+    modifyParts: setCustomSymbol('Ξ'),
+    getFormatterConfig: memoize((value) => {
+      return {
```

Comment:
let's create this object statically so that we're returning the same object for different values, too

Reply:
Ok!